drivers/sensor/ti/tmp114/tmp114.c

```diff
@@ -161,11 +191,47 @@ static int tmp114_attr_get(const struct device *dev, enum sensor_channel chan,
 	return 0;
 }
 
+static int16_t tmp114_odr_value(const struct sensor_value *frequency)
+{
+	const uint32_t freq_micro = sensor_value_to_micro(frequency);
+	int conv = 0;
+
+	switch (freq_micro) {
+	case 156000000:
+		conv = TMP114_DT_ODR_6_4_MS; /* 6.4 ms */
+		break;
+	case 32000000:
+		conv = TMP114_DT_ODR_31_25_MS; /* 32.25 ms */
+		break;
+	case 16000000:
+		conv = TMP114_DT_ODR_62_5_MS; /* 62.5 ms */
+		break;
+	case 8000000:
+		conv = TMP114_DT_ODR_125_MS; /* 125 ms */
+		break;
+	case 4000000:
+		conv = TMP114_DT_ODR_250_MS; /* 250 ms */
+		break;
+	case 2000000:
+		conv = TMP114_DT_ODR_500_MS; /* 500 ms */
+		break;
+	case 1000000:
+		conv = TMP114_DT_ODR_1000_MS; /* 1 s */
+		break;
+	case 500000:
+		conv = TMP114_DT_ODR_2000_MS; /* 2 s */
+		break;
+	default:
+		return -EINVAL;
+	}
+	return conv;
+}
+
 static int tmp114_attr_set(const struct device *dev, enum sensor_channel chan,
 			   enum sensor_attribute attr, const struct sensor_value *val)
 {
-	int16_t value;
-	int rc;
+	int16_t avg = 0;
+	uint16_t conv = 0;
```

Comment:
A shared generic `value` more is more scalable if/when more cases are added to the switch. I also think it aligns with [Linux naming coding style](https://www.kernel.org/doc/html/latest/process/coding-style.html#naming).
Changing it from `int16_t` to `uint16_t` seems correct.